File: NC/longest-consecutive-sequence.py
class Solution:

    def getLongestSequence(self, numSet, numTraversed, num):
        if numTraversed.get(num, False) == True:
            return 1

        ans = 1

        numTraversed[num] = True

        curr = num - 1

        while curr in numSet:
            ans += 1
            numTraversed[curr] = True
            curr -= 1

        curr = num + 1

        while curr in numSet:
            ans += 1
            numTraversed[curr] = True
            curr += 1

        return ans

    def longestConsecutive(self, nums: List[int]) -> int:
        numSet = set()
        numTraversed = {}
        for num in nums:
            numSet.add(num)

        ans = 0

        for num in numSet:
            ans = max(ans, self.getLongestSequence(numSet, numTraversed, num))

        return ans

File: NC/java/LongestConsecutiveSequence.java
package NC.java;

import java.util.Arrays;
import java.util.HashSet;
import java.util.Set;

public class LongestConsecutiveSequence {

    public int longestConsecutive(int[] nums) {
        Set<Integer> numSet = new HashSet();
        Set<Integer> visited = new HashSet();

        for (int num : nums) {
            numSet.add(num);
        }

        int ans = 0;

        for (int num : nums) {
            if (visited.contains(num)) {
                continue;
            } else {
                visited.add(num);
            }
            int lowest = num;
            while (numSet.contains(lowest - 1)) {
                lowest--;
                visited.add(lowest);
            }
            int highest = num;
            while (numSet.contains(highest + 1)) {
                highest++;
                visited.add(highest);
            }
            ans = Math.max(ans, highest - lowest + 1);
        }

        return ans;

    }
}
